LDMP/jobs/manager.py

```diff
@@ -98,28 +99,48 @@ def _get_extent_tuple_raster(path):
 def _get_extent_tuple_vector(path):
     if conf.settings_manager.get_value(conf.Setting.DEBUG):
         log(f"Trying to calculate extent of vector {path}")
-    rect = QgsVectorLayer(str(path), "vector file", "ogr").extent()
-    if rect:
-        xmin = rect.xMinimum()
-        xmax = rect.xMaximum()
-        ymin = rect.yMinimum()
-        ymax = rect.yMaximum()
-        if any([(val > 180 or val < -180) for val in [xmin, xmax]]) or any(
-            [(val > 90 or val < -90) for val in [ymin, ymax]]
-        ):
-            # If there are not yet any features, then the extent will be set as ranging
-            # from -infinite to infinite - so catch this with above check
-            if conf.settings_manager.get_value(conf.Setting.DEBUG):
-                log(f"Failed to calculate extent for {path} - appears undefined")
-            return None
-        else:
-            if conf.settings_manager.get_value(conf.Setting.DEBUG):
-                log(f"Calculated extent for {path} - {(xmin, ymin, xmax, ymax)}")
-            return (xmin, ymin, xmax, ymax)
-    else:
-        log("Failed to calculate extent - couldn't open dataset")
+
+    layer = QgsVectorLayer(str(path), "vector file", "ogr")
+    rect = layer.extent()
+
+    try:
+        is_empty = hasattr(rect, "isEmpty") and rect.isEmpty()
+    except Exception:
+        is_empty = False
+
+    if (not rect) or is_empty:
+        if conf.settings_manager.get_value(conf.Setting.DEBUG):
+            log(f"Failed to calculate extent for {path} - extent empty/undefined")
         return None
 
+    xmin = rect.xMinimum()
+    xmax = rect.xMaximum()
+    ymin = rect.yMinimum()
+    ymax = rect.yMaximum()
+
+    vals = [xmin, ymin, xmax, ymax]
+    if any(
+        (v is None) or (isinstance(v, float) and (math.isnan(v) or math.isinf(v)))
+        for v in vals
+    ):
+        if conf.settings_manager.get_value(conf.Setting.DEBUG):
+            log(f"Failed to calculate extent for {path} - NaN/Inf in extent {vals}")
+        return None
+
+    if any((val > 180 or val < -180) for val in [xmin, xmax]) or any(
+        (val > 90 or val < -90) for val in [ymin, ymax]
+    ):
+        if conf.settings_manager.get_value(conf.Setting.DEBUG):
+            log(
+                f"Failed to calculate extent for {path} - appears undefined/out of bounds "
+                f"({xmin}, {ymin}, {xmax}, {ymax})"
+            )
+        return None
```

Comment:
This check won't work in all cases - projected coordinate systems will have xmin/xmax/ymin/ymax outside these bounds. This is an existing bug so fine to keep this check in for now so we can close this issue, but we should create a new issue and fix for this down the road with a more resilient check of the values

Reply:
thanks for the input @azvoleff